File: Structural/Adapter/CSharp/VLCPlayer.cs
// VLCPlayer.cs
using System;

public class VLCPlayer : IAdvancedMediaPlayer {
  public void PlayMP4(string fileName) {
    // Do nothing, VLCPlayer does not support MP4
  }

  public void PlayVLC(string fileName) {
    Console.WriteLine("Playing VLC file: " + fileName);
  }
}

File: Structural/Adapter/Java/VLCPlayer.java
// VLCPlayer.java
public class VLCPlayer implements AdvancedMediaPlayer {
  @Override
  public void playMP4(String fileName) {
    // Do nothing, VLCPlayer does not support MP4
  }

  @Override
  public void playVLC(String fileName) {
    System.out.println("Playing VLC file: " + fileName);
  }
}
